doc/globals.rdoc

```diff
@@ -1,69 +1,425 @@
+
+Some of the pre-defined global variables have synonyms
+that are available via module Engish.
+For each of those, the \English synonym is given.
+
+To use the module:
+
+  require 'English'
+
+=== Exceptions
+
+==== <tt>$!</tt> (\Exception)
+
+Contains the Exception object set by Kernel#raise:
+
+  begin
+    raise RuntimeError.new('Boo!')
+  rescue RuntimeError
+    p $!
+  end
+
+Output:
+
+  #<RuntimeError: Boo!>
+
+English - <tt></tt>ERROR_INFO
+
+==== <tt>$@</tt> (Backtrace)
+
+Same as <tt>$!.backtrace</tt>;
+returns an array of backtrace positions:
+
+  begin
+    raise RuntimeError.new('Boo!')
+  rescue RuntimeError
+    pp $@.take(4)
+  end
+
+Output:
+
+  ["(irb):338:in `<top (required)>'",
+   "/snap/ruby/317/lib/ruby/3.2.0/irb/workspace.rb:119:in `eval'",
+   "/snap/ruby/317/lib/ruby/3.2.0/irb/workspace.rb:119:in `evaluate'",
+   "/snap/ruby/317/lib/ruby/3.2.0/irb/context.rb:502:in `evaluate'"]
+
+English - $ERROR_POSITION
+
+=== Pattern Matching
+
+These global variables store information about the most recent
+successful match in the current scope.
+
+For details and examples,
+see {Regexp Global Variables}[rdoc-ref:Regexp@Global+Variables].
+
+==== <tt>$~</tt> (\MatchData)
+
+MatchData object created from the match;
+thread-local and frame-local.
+
+English - <tt>$LAST_MATCH_INFO</tt>.
+
+==== <tt>$&</tt> (Matched Substring)
+
+The matchedstring.
+
+English - <tt>$MATCH</tt>.
+
+==== <tt>$`</tt> (Pre-Match Substring)
+
+The string to the left of the match.
+
+English - <tt>$PREMATCH</tt>.
+
+==== <tt>$'</tt> (Post-Match Substring)
+
+The string to the right of the match.
+
+English - <tt>$POSTMATCH</tt>.
+
+==== <tt>$+</tt> (Last Matched Group)
+
+The last group matched.
+
+English - <tt>$LAST_PAREN_MATCH</tt>.
+
+==== <tt>$1</tt>, <tt>$2</tt>, \Etc. (Matched Group)
+
+For <tt>$_n_</tt> the _nth_ group of the match.
+
+No \English.
+
+=== Separators
+
+==== <tt>$/</tt> (Input Record Separator)
+
+An input record separator, initially newline.
+
+English - <tt>$INPUT_RECORD_SEPARATOR</tt>, <tt>$RS</tt>.
+
+Aliased as <tt>$-0</tt>.
+
+==== <tt>$;</tt> (Input Field Separator)
+
+An input field separator, initially +nil+.
+
+English - <tt>$FIELD_SEPARATOR</tt>, <tt>$FS</tt>.
+
+Aliased as <tt>$-F</tt>.
+
+==== <tt>$\\</tt> (Output Record Separator)
+
+An output record separator, initially +nil+.
+
+English - <tt>$OUTPUT_RECORD_SEPARATOR</tt>, <tt>$ORS</tt>.
+
+==== <tt>$,</tt> (Output Field Separator)
+
+An output field separator, initially +nil+.
+
+English - <tt>$OUTPUT_FIELD_SEPARATOR</tt>, <tt>$OFS</tt>.
+
+=== Streams
+
+==== <tt>$stdin</tt> (Standard Input)
+
+The current standard input stream; initially:
+
+  $stdin # => #<IO:<STDIN>>
+
+==== <tt>$stdout</tt> (Standard Output)
+
+The current standard output stream; initially:
+
+  $stdout # => #<IO:<STDOUT>>
+
+==== <tt>$stderr</tt> (Standard Error)
+
+The current standard error stream; initially:
+
+  $stderr # => #<IO:<STDERR>>
+
+==== <tt>$</tt> (Default Standard Input)
```

Comment:
Do you mean `$<`? I'm also not sure if "Default Standard Input" is the right term for this. `ARGF` is used to operate on files passed as command-line arguments (or stdin if there are no files passed in).